core/rawdb/chain_iterator.go

```diff
@@ -135,32 +135,14 @@ func iterateTransactions(db ethdb.Database, from uint64, to uint64, reverse bool
 				close(hashesCh)
 			}
 		}()
-
-		var hasher = sha3.NewLegacyKeccak256()
 		for data := range rlpCh {
-			it, err := rlp.NewListIterator(data.rlp)
-			if err != nil {
-				log.Warn("tx iteration error", "error", err)
-				return
-			}
-			it.Next()
-			txs := it.Value()
-			txIt, err := rlp.NewListIterator(txs)
-			if err != nil {
-				log.Warn("tx iteration error", "error", err)
-				return
+			var body types.Body
+			if err := rlp.DecodeBytes(data.rlp, &body); err != nil {
+				log.Warn("tx iteration error [1]", "block", data.number, "error", err)
```

Comment:
Can we rename the warning message that: `Failed to decode the block body`?

Reply:
And also originally, the procedure is aborted whenever error happens, now only the warning log is printed and the error is skipped.

The `iterateTransactions` will be used in transaction indexing and unindexing. It will lead to the data loss eventually(e.g. the transaction indexes are missing). Not sure it's safe enough, although the transaction indexes are not very important.

Reply:
Good catch, fixed